TestRedClusterCanBeModifiedByDisablingPredicate is flaky

`TestRedClusterCanBeModifiedByDisablingPredicate` failed on `GKE / 7.17.x` with Stack `7.17.6`:

```
[2022-11-21T03:35:42.230Z] === Failed

[2022-11-21T03:35:42.230Z] === FAIL: test/e2e/es TestRedClusterCanBeModifiedByDisablingPredicate/Misconfigure_index_on_cluster,_turning_cluster_red (0.00s)

[2022-11-21T03:35:42.230Z]     disable_predicates_test.go:76: 

[2022-11-21T03:35:42.230Z]         	Error Trace:	/go/src/github.com/elastic/cloud-on-k8s/test/e2e/es/disable_predicates_test.go:76

[2022-11-21T03:35:42.230Z]         	Error:      	Received unexpected error:

[2022-11-21T03:35:42.230Z]         	            	elasticsearch client failed for https://test-v-up-with-red-cluster-7x-wvsj-es-internal-http.e2e-qpoza-mercury.svc:9200/test-index: Put "https://test-v-up-with-red-cluster-7x-wvsj-es-internal-http.e2e-qpoza-mercury.svc:9200/test-index": dial tcp 10.72.172.140:9200: connect: connection refused

[2022-11-21T03:35:42.230Z]         	Test:       	TestRedClusterCanBeModifiedByDisablingPredicate/Misconfigure_index_on_cluster,_turning_cluster_red

[2022-11-21T03:35:42.230Z] {"log.level":"error","@timestamp":"2022-11-21T03:33:58.363Z","message":"stopping early","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0","error":"test failure","error.stack_trace":"github.com/elastic/cloud-on-k8s/v2/test/e2e/test.StepList.RunSequential\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/step.go:45\ngithub.com/elastic/cloud-on-k8s/v2/test/e2e/es.TestRedClusterCanBeModifiedByDisablingPredicate\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/es/disable_predicates_test.go:96\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1446"}

[2022-11-21T03:35:42.230Z]     --- FAIL: TestRedClusterCanBeModifiedByDisablingPredicate/Misconfigure_index_on_cluster,_turning_cluster_red (0.00s)
```

--- a/test/e2e/es/disable_predicates_test.go
+++ b/test/e2e/es/disable_predicates_test.go
@@ -12,8 +12,6 @@ import (
 	"net/http"
 	"testing"
 
-	"github.com/stretchr/testify/require"
-
 	esv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/elasticsearch/v1"
 	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/elasticsearch/client"
 	"github.com/elastic/cloud-on-k8s/v2/test/e2e/test"
@@ -54,15 +52,18 @@ func TestRedClusterCanBeModifiedByDisablingPredicate(t *testing.T) {
 			elasticsearch.CheckClusterHealth(initial, k),
 			{
 				Name: "Create ES client",
-				Test: func(t *testing.T) {
+				Test: test.Eventually(func() error {
 					var err error
 					esClient, err = elasticsearch.NewElasticsearchClient(initial.Elasticsearch, k)
-					require.NoError(t, err)
-				},
+					if err != nil {
+						return err
+					}
+					return nil
+				}),
 			},
 			{
 				Name: "Misconfigure index on cluster, turning cluster red",
-				Test: func(t *testing.T) {
+				Test: test.Eventually(func() error {
 					settings := `
 {
     "settings": {
@@ -71,11 +72,17 @@ func TestRedClusterCanBeModifiedByDisablingPredicate(t *testing.T) {
 }
 `
 					r, err := http.NewRequest(http.MethodPut, "/test-index", bytes.NewBufferString(settings))
-					require.NoError(t, err)
+					if err != nil {
+						return err
+					}
 					response, err := esClient.Request(context.Background(), r)
-					require.NoError(t, err)
+					if err != nil {
+						return err
+					}
 					defer response.Body.Close() // nolint
-				},
+
+					return nil
+				}),
 			},
 			// wait for the cluster to become red
 			elasticsearch.CheckSpecificClusterHealth(initial, k, esv1.ElasticsearchRedHealth),
@@ -84,13 +91,18 @@ func TestRedClusterCanBeModifiedByDisablingPredicate(t *testing.T) {
 		[]test.Step{
 			{
 				Name: "Delete misconfigured index on cluster, allowing cluster to turn back green",
-				Test: func(t *testing.T) {
+				Test: test.Eventually(func() error {
 					r, err := http.NewRequest(http.MethodDelete, "/test-index", nil)
-					require.NoError(t, err)
+					if err != nil {
+						return err
+					}
 					response, err := esClient.Request(context.Background(), r)
-					require.NoError(t, err)
+					if err != nil {
+						return err
+					}
 					defer response.Body.Close() // nolint
-				},
+					return nil
+				}),
 			},
 		},
 	).RunSequential(t)

